tests/TestSymmetricCipher.cpp

```diff
@@ -32,6 +32,98 @@ void TestSymmetricCipher::initTestCase()
     QVERIFY(Crypto::init());
 }
 
+void TestSymmetricCipher::testAes128CbcEncryption()
+{
+    // http://csrc.nist.gov/publications/nistpubs/800-38a/sp800-38a.pdf
+
+    QByteArray key = QByteArray::fromHex("2b7e151628aed2a6abf7158809cf4f3c");
+    QByteArray iv = QByteArray::fromHex("000102030405060708090a0b0c0d0e0f");
+    QByteArray plainText = QByteArray::fromHex("6bc1bee22e409f96e93d7e117393172a");
+    plainText.append(QByteArray::fromHex("ae2d8a571e03ac9c9eb76fac45af8e51"));
+    QByteArray cipherText = QByteArray::fromHex("7649abac8119b246cee98e9b12e9197d");
+    cipherText.append(QByteArray::fromHex("5086cb9b507219ee95db113a917678b2"));
+    bool ok;
+
+    SymmetricCipher cipher(SymmetricCipher::Aes128, SymmetricCipher::Cbc, SymmetricCipher::Encrypt);
+    QVERIFY(cipher.init(key, iv));
+    QCOMPARE(cipher.blockSize(), 16);
+
+    QCOMPARE(cipher.process(plainText, &ok),
+             cipherText);
+    QVERIFY(ok);
+
+    QBuffer buffer;
+    SymmetricCipherStream stream(&buffer, SymmetricCipher::Aes128, SymmetricCipher::Cbc,
+                                 SymmetricCipher::Encrypt);
+    QVERIFY(stream.init(key, iv));
+    buffer.open(QIODevice::WriteOnly);
+    QVERIFY(stream.open(QIODevice::WriteOnly));
+    QVERIFY(stream.reset());
+
+    buffer.reset();
+    buffer.buffer().clear();
+    QCOMPARE(stream.write(plainText.left(16)), qint64(16));
+    QCOMPARE(buffer.data(), cipherText.left(16));
+    QVERIFY(stream.reset());
+    // make sure padding is written
+    QCOMPARE(buffer.data().size(), 32);
+
+    buffer.reset();
+    buffer.buffer().clear();
+    QCOMPARE(stream.write(plainText.left(10)), qint64(10));
+    QVERIFY(buffer.data().isEmpty());
+
+    QVERIFY(stream.reset());
+    buffer.reset();
+    buffer.buffer().clear();
+    QCOMPARE(stream.write(plainText.left(10)), qint64(10));
+    stream.close();
+    QCOMPARE(buffer.data().size(), 16);
+}
+
+void TestSymmetricCipher::testAes128CbcDecryption()
+{
+    QByteArray key = QByteArray::fromHex("2b7e151628aed2a6abf7158809cf4f3c");
+    QByteArray iv = QByteArray::fromHex("000102030405060708090a0b0c0d0e0f");
+    QByteArray cipherText = QByteArray::fromHex("7649abac8119b246cee98e9b12e9197d");
+    cipherText.append(QByteArray::fromHex("5086cb9b507219ee95db113a917678b2"));
+    QByteArray plainText = QByteArray::fromHex("6bc1bee22e409f96e93d7e117393172a");
+    plainText.append(QByteArray::fromHex("ae2d8a571e03ac9c9eb76fac45af8e51"));
+    bool ok;
+
+    SymmetricCipher cipher(SymmetricCipher::Aes128, SymmetricCipher::Cbc, SymmetricCipher::Decrypt);
+    QVERIFY(cipher.init(key, iv));
+    QCOMPARE(cipher.blockSize(), 16);
+
+    QCOMPARE(cipher.process(cipherText, &ok),
+             plainText);
+    QVERIFY(ok);
+
+    // padded with 16 0x16 bytes
+    QByteArray cipherTextPadded = cipherText + QByteArray::fromHex("3a3aa5e0213db1a9901f9036cf5102d2");
```

Comment:
You are trying to pad the cipher text here, which is incorrect and also makes no sense. `cipherText + QByteArray::fromHex("3a3aa5e0213db1a9901f9036cf5102d2")` is the cipher text for `plainText` + padding when encrypted with AES256 and the key `603deb1015ca71be2b73aef0857d77811f352c073b6108d72d9810a30914dff4`. For AES128 with the key `2b7e151628aed2a6abf7158809cf4f3c`, the cipher text is of course different.

The padding scheme appears to be PKCS#5, which means you need to add N padding bytes of value N to the _plaintext_. In case of a full block, this is 16 padding bytes of value 16 (which is 0x10 in hex, the comment is misleading).